src/recordlinker/routes/person_router.py

```diff
@@ -75,3 +100,58 @@ def update_person(
 
     person = service.update_person_cluster(session, patients, person, commit=False)
     return schemas.PersonRef(person_reference_id=person.reference_id)
+
+
+def persons_by_reference_ids_or_422(
+    session: orm.Session, reference_ids: typing.Sequence[uuid.UUID]
+) -> typing.Sequence[models.Person]:
+    """
+    Retrieve the Persons by their reference ids or raise a 422 error response.
+    """
+    persons = service.get_persons_by_reference_ids(session, reference_ids)
+    if not persons:
+        raise fastapi.HTTPException(
+            status_code=fastapi.status.HTTP_422_UNPROCESSABLE_ENTITY,
+            detail=[
+                {
+                    "loc": ["body", "persons"],
+                    "msg": "Invalid person_reference_id(s). No persons found for any of the provided person_rederence_id(s)",
+                    "type": "value_error",
+                }
+            ],
+        )
+    return persons  # type: ignore
+
+
+@router.post(
+    "/{person_reference_id}/merge",
+    summary="Merge Person clusters",
+    status_code=fastapi.status.HTTP_200_OK,
+)
+def merge_person_clusters(
+    person_reference_id: uuid.UUID,
+    data: typing.Annotated[schemas.PersonRefs, fastapi.Body()],
+    session: orm.Session = fastapi.Depends(get_session),
+) -> schemas.PersonRef:
+    """
+    Merges Person cluster(s) into the Person cluster referenced by `person_reference_id`.
+    """
+    # Get the person_reference_id the person clusters will be merged into
+    person = service.get_person_by_reference_id(session, person_reference_id)
+
+    if person is None:
+        raise fastapi.HTTPException(status_code=fastapi.status.HTTP_404_NOT_FOUND)
+
+    # Get all persons by person_reference_id that will be merged
+    persons = persons_by_reference_ids_or_422(session, data.person_reference_ids)
+    person_ids = [person.id for person in persons]
+
+    # Get all of the patients from the person clusters to be merged
+    patients = patients_by_person_id_or_422(session, person_ids)
+
+    # Merge the person clusters
+    person = service.update_person_cluster(session, patients, person, commit=False)
```

Comment:
If you wanted to do this in one query rather than 2 (merge lines 150 and 153), I think you could write a mpi_service function to do that.
```
UPDATE patients p
SET p.person_id = X
WHERE p.person_id in (Y, Z ,...)
```

Reply:
I like that and will update accordingly!